zulipterminal/ui_tools/views.py

```diff
@@ -536,11 +537,12 @@ def update_user_list(self, search_box: Any=None,
             if user_list:
                 self.view.users = user_list
             users = self.view.users.copy()
-            users_display = users.copy()
             if new_text:
-                for user in users:
-                    if not match_user(user, new_text):
-                        users_display.remove(user)
+                users_display = [
+                    user for user in users if match_user(user, new_text)
+                ]
+            else:
+                users_display = users
```

Comment:
Would `users_to_display` be a slightly better name for this?

Reply:
I was focusing on adjusting the logic in the first instance.